data_structures/queue/priority_queue.py

```diff
@@ -0,0 +1,273 @@
+            raise UnderFlow("Under Flow!")
+
+        if len(self.queue[0]) != 0:
+            return self.queue[0].pop(0)
+        elif len(self.queue[1]) != 0:
+            return self.queue[1].pop(0)
+        else:
+            return self.queue[2].pop(0)
+
+    def print_queue(self):
+        print("Priority 0:", self.queue[0])
+        print("Priority 1:", self.queue[1])
+        print("Priority 2:", self.queue[2])
+
+    def __str__(self):
+        """
+        Prints each priority queue within the FixedPriorityQueue
+        """
+        s = ""
+        for i in range(len(self.queue)):
+            for j in self.queue[i]:
+                s += "Priority " + str(i) + ": "
+                s += str(i) + " "
+            s += "\n"
+        print(s)
+        return s
+
+
+class ElementPriorityQueue:
+    """
+    Element Priority Queue is the same as Fixed Priority Queue
+    The only difference being the value of the element itself is the priority
+    The rule for priorities are the same
+    The lowest integer has the highest priority.
+    Example:
+    priority(0) > priority(5)
+    priority(16) > priority(32)
+    You can change the priorities as per your need
+
+    >>> EPQ = ElementPriorityQueue()
+    >>> EPQ.enqueue(10)
+    >>> EPQ.enqueue(70)
+    >>> EPQ.enqueue(100)
+    >>> EPQ.enqueue(1)
+    >>> EPQ.enqueue(5)
+    >>> EPQ.enqueue(7)
+    >>> EPQ.enqueue(4)
+    >>> EPQ.enqueue(64)
+    >>> EPQ.enqueue(128)
+    >>> EPQ.print_queue()
+    [10, 70, 100, 1, 5, 7, 4, 64, 128]
+    >>> EPQ.dequeue()
+    1
+    >>> EPQ.dequeue()
+    4
+    >>> EPQ.dequeue()
+    5
+    >>> EPQ.dequeue()
+    7
+    >>> EPQ.dequeue()
+    10
+    >>> EPQ.print_queue()
+    [70, 100, 64, 128]
+    >>> EPQ.dequeue()
+    64
+    >>> EPQ.dequeue()
+    70
+    >>> EPQ.dequeue()
+    100
+    >>> EPQ.dequeue()
+    128
+    >>> EPQ.dequeue()
+    Traceback (most recent call last):
+                    ...
+    priority_queue.UnderFlow: Under Flow!
+    """
+
+    def __init__(self):
+        self.queue = []
+
+    def enqueue(self, data):
+        """
+        This function enters the element into the queue
+        If the queue is full an Exception is raised saying Over Flow!
+        """
+        if len(self.queue) == 100:
+            raise OverFlow("Over Flow!")
+        else:
+            self.queue.append(data)
+
+    def dequeue(self):
+        """
+        Return the highest priority element in FIFO order.
+        If the queue is empty then an under flow exception is raised.
+        """
+        if len(self.queue) == 0:
+            raise UnderFlow("Under Flow!")
+        else:
+            data = min(self.queue)
+            self.queue.remove(data)
+            return data
+
+    def print_queue(self):
+        print(self.queue)
+
+    def __str__(self):
+        """
+        Prints all the elements within the Element Priority Queue
+        """
+        s = ""
+        for i in self.queue:
```

Comment:
There are built-in ways of doing this. Firstly you could do this:

```python
>>> queue = [1, 2, 3, 4, 5]
>>> ' '.join(str(item) for item in queue)
'1 2 3 4 5'
```

You could also just do this
```python
>>> queue = [1, 2, 3, 4, 5]
>>> f'{queue!s}'
'[1, 2, 3, 4, 5]'
```
python will do a lot for you, if you let it.

Reply:
OH!
The second snippet is something new to me